### DISCUSSION TITLE
Synchronization and Projectwise WSG

### DISCUSSION THREAD

**Discussion by inigolorente** [iTwin/itwinjs-core#4871] [Q&A]
Hello,

After reading the documentation, I am not sure how can I connect an iTwin with the Synchronization API to models hosted on ProjectWise (with the WSG plugin activated).
How can I get an IMS token that can be used with the ProjectWise WSG server to fetch the models and synchronize them?

Thanks in advance for your replies :)

**Comment by arnab-git-hub** [comment]
I had contacted our ProjectWise WSG dev regarding this issue who mentioned OAuth2 feature is not possible right now and will be available in the future.

**Comment by Aidas-Vaiksnoras** [comment]
Hello, I am Aidas from the Synchronization API team. 
As @ArnabGogoi replied, ProjectWise WSG servers do not support OAuth2 tokens, so it cannot be accessed with IMS client tokens.

**Comment by candcconsulting** [comment]
@inigolorente did you see this article ?
https://communities.bentley.com/products/digital-twin-cloud-services/itwin-services/w/synchronization-wiki/64014/how-to-build-your-own-synchronizer
